hera_pspec/pspecdata.py

```diff
@@ -708,31 +708,37 @@ def p_hat(self, M, q):
             Optimal estimate of bandpower, \hat{p}.
         """
         return np.dot(M, q)
-    
-    def units(self):
+
+    def units(self, little_h=True):
         """
         Return the units of the power spectrum. These are inferred from the 
         units reported by the input visibilities (UVData objects).
 
+        Parameters
+        ----------
+        little_h : boolean, optional
+                Whether to have cosmological length units be h^-1 Mpc or Mpc
+                Default: h^-1 Mpc
+
         Returns
         -------
         pspec_units : str
             Units of the power spectrum that is returned by pspec().
-        
-        delay_units : str
-            Units of the delays (wavenumbers) returned by pspec().
         """
-        # Frequency units of UVData are always Hz => always convert to ns
-        delay_units = 'ns'
-        
         # Work out the power spectrum units
         if len(self.dsets) == 0:
             raise IndexError("No datasets have been added yet; cannot "
                              "calculate power spectrum units.")
+        if self.primary_beam is None:
+            pspec_units = "improper normalization"
```

Comment:
Is there something a little more informative that we could write here? e.g. `Beam or normalization not specified`?

Reply:
sure!